extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts

```diff
@@ -49,5 +65,136 @@ export class DataExtensionsEditorView extends AbstractWebview<
 
   protected async onWebViewLoaded() {
     super.onWebViewLoaded();
+
+    await this.loadExternalApiUsages();
+  }
+
+  protected async loadExternalApiUsages(): Promise<void> {
+    const queryResult = await this.runQuery();
+    if (!queryResult) {
+      await this.clearProgress();
+      return;
+    }
+
+    await this.showProgress({
+      message: "Loading results",
+      step: 1100,
+      maxStep: 1500,
+    });
+
+    const bqrsPath = queryResult.outputDir.bqrsPath;
+
+    const results = await this.getResults(bqrsPath);
+    if (!results) {
+      await this.clearProgress();
+      return;
+    }
+
+    await this.showProgress({
+      message: "Finalizing results",
+      step: 1450,
+      maxStep: 1500,
+    });
+
+    await this.postMessage({
+      t: "setExternalApiRepoResults",
+      results,
+    });
+
+    await this.clearProgress();
+  }
+
+  private async runQuery(): Promise<CoreCompletedQuery | undefined> {
+    const qlpacks = await qlpackOfDatabase(this.cliServer, this.databaseItem);
+
+    const packsToSearch = [qlpacks.dbschemePack];
+    if (qlpacks.queryPack) {
+      packsToSearch.push(qlpacks.queryPack);
+    }
+
+    const suiteFile = (
+      await file({
+        postfix: ".qls",
+      })
+    ).path;
+    const suiteYaml = [];
+    for (const qlpack of packsToSearch) {
+      suiteYaml.push({
```

Comment:
I don't think it's worth it. We're only using this in two places (the contextual query resolver and here), and the format is quite complex: https://docs.github.com/en/code-security/codeql-cli/using-the-codeql-cli/creating-codeql-query-suites. We're only using a very limited subset of this, and I think it would be more confusing than helpful if you try to add something to this which is valid, but not in our types.